forge/src/coverage.rs

```diff
@@ -0,0 +1,224 @@
+use comfy_table::{Attribute, Cell, Color, Row, Table};
+pub use foundry_evm::coverage::*;
+use std::{collections::HashMap, io::Write, path::PathBuf};
+
+/// A coverage reporter.
+pub trait CoverageReporter {
+    fn build(&mut self, map: CoverageMap);
+    fn finalize(self) -> eyre::Result<()>;
+}
```

Comment:
looking at the `match self.report` earlier

we don't really make use of that, right?

but nice to have